k2/csrc/fsa_algo.cc

```diff
@@ -383,6 +419,111 @@ void RmEpsilonsPrunedMax(const WfsaWithFbWeights &a, float beam, Fsa *b,
   b->arc_indexes.push_back(b->arc_indexes.back());
 }
 
+void RmEpsilonsPrunedLogSum(
+    const WfsaWithFbWeights &a, float beam, Fsa *b,
+    std::vector<float> *b_arc_weights,
```

Comment:
Shoul we add arc_weights_out for Max version, otherwise I will add a function to get arc_weights from `arc_map` or `arc_derivs` (max version) later.

Reply:
No, don't have arc_weights_out for Max; easier for autograd if the python code does that.  
For testing purposes you could add a function that does it at the C++ level too, though.

Reply:
Ok, sure